pkg/cmd/server/origin/master.go

```diff
@@ -249,6 +250,15 @@ func (c *MasterConfig) Run(kubeAPIServerConfig *kubeapiserver.Config, controller
 	// add post-start hooks
 	aggregatedAPIServer.GenericAPIServer.AddPostStartHookOrDie("template.openshift.io-sharednamespace", c.ensureOpenShiftSharedResourcesNamespace)
 	aggregatedAPIServer.GenericAPIServer.AddPostStartHookOrDie("authorization.openshift.io-bootstrapclusterroles", bootstrappolicy.Policy().EnsureRBACPolicy())
+	aggregatedAPIServer.GenericAPIServer.AddPostStartHookOrDie("admission.openshift.io-RefreshRESTMapper", func(context apiserver.PostStartHookContext) error {
+		c.RESTMapper.Reset()
+		go func() {
+			wait.Until(func() {
+				c.RESTMapper.Reset()
+			}, 10*time.Second, context.StopCh)
```

Comment:
So it fails well, doesn't crash the process, doesn't affect apiserver response time, and handles partials as well as can be expected.  Anyone want to come out with an "I prefer to wait longer, so please change it to value X"?